### ISSUE TITLE
When starting game stream with multiple controllers, players are connected in reverse order

### ISSUE THREAD

**Issue by teconmoon** [moonlight-stream/moonlight-qt#568]
**Likely Irrelevant background information**: I have four DualShock 3 Controllers that I use with moonlight-qt on what’s basically a moonlight thin client. I use Debian instead of Windows for this thin client due to drastically superior DS3 drivers built-in to the Linux kernel for a long time, whereas Windows DS3 controller support is still a struggle to this day. (Getting better with various ViGEm projects)

**Problem**: I’ve observed that if I connect multiple game controllers _before_ starting a game stream, once the stream starts they’ll be connected to the host consistently in the reverse order, the end result being that Players 1 2 3 4 ends up being 4 3 2 1. The same occurs if playing with only two or three controllers, 1 2 3 becomes 3 2 1, etc.

If I connect the controllers _after_ the game stream starts, they are connected in the correct order. This workaround is simple enough to remember, but has some limitations. For example:
1. I start a game stream
2. I connect the controllers in order (yay)
3. We exit the game stream to launch a different game
4. Upon the new stream starting, our controller order is reversed, causing confusion

Right now the most feasible workaround is just turning on the controllers in reverse order that they’ll be used in ahead of time.

Thanks for reading! Not sure they’re relevant aside from OS and client version but specs below.

Client: [Minisforum N40 Mini PC](https://store.minisforum.com/products/minisforum-n40-mini-pc) 
OS: Debian 10 (KDE Plasma)
CPU: Intel Celeron N4000
GPU: Intel UHD Graphics 600
App version: v3.1.1 (Snap)

Host:
OS: Windows 10 Enterprise LTSC 2019
CPU: Intel Xeon E5-2696 v2
GPU: MSI GeForce GTX 1650 D6 GAMING X
Driver: 465.89
GFE: 3.21.0.36

**Comment by cgutman** [comment]
I suspect this is a bug in SDL. We simply assign players IDs in the order we see the controllers appear.

Can you try the Flatpak package? SDL uses different controller enumeration logic in a Flatpak container because udev is not available.

**Comment by teconmoon** [comment]
Thank you for your reply! I have now tested Flatpak and am seeing the same behavior.

**Comment by teconmoon** [comment]
I also tested the AppImage today and what I saw was that it still behaved oddly, though not exactly the same problem as snap/flatpak.

When connecting 4 controllers, 1234 became 2341 instead of the 4321 reverse behavior I saw prior.

I’m not sure what the differences between versions are in controller handling, but I’ll do more testing. Snap / FlatPak behavior was 100% consistent so I’ll see if this new AppImage behavior is also 100% consistent.
